qui/updater/updater.py

```diff
@@ -0,0 +1,216 @@
+        self.cancel_button = self.builder.get_object("button_cancel")
+        self.cancel_button.connect("clicked", self.cancel_clicked)
+
+        load_theme(widget=self.main_window,
+                   light_theme_path=pkg_resources.resource_filename(
+                       'qui', 'qubes-updater-light.css'),
+                   dark_theme_path=pkg_resources.resource_filename(
+                       'qui', 'qubes-updater-dark.css'))
+        self.theme = Theme.LIGHT if is_theme_light(self.main_window) \
+            else Theme.DARK
+
+        self.header_label = self.builder.get_object("header_label")
+
+        self.intro_page = IntroPage(
+            self.builder, self.theme, self.next_button)
+        self.progress_page = ProgressPage(
+            self.builder,
+            self.theme,
+            self.header_label,
+            self.next_button,
+            self.cancel_button
+        )
+        self.summary_page = SummaryPage(
+            self.builder,
+            self.theme,
+            self.next_button,
+            self.cancel_button,
+            self.progress_page.back_by_row_selection
+        )
+
+
+        self.button_settings = self.builder.get_object("button_settings")
+        self.button_settings.connect("clicked", self.open_settings_window)
+        settings_pixbuf = load_icon_at_gtk_size(
+            'qubes-customize', Gtk.IconSize.LARGE_TOOLBAR)
+        settings_image = Gtk.Image.new_from_pixbuf(settings_pixbuf)
+        self.button_settings.set_image(settings_image)
+        self.settings = Settings(
+            self.main_window,
+            self.qapp,
+            refresh_callback=self.intro_page.refresh_update_list
+        )
+
+        self.intro_page.populate_vm_list(self.qapp, self.settings)
+
+        headers = [(3, "intro_name"), (3, "progress_name"), (3, "summary_name"),
+                   (3, "restart_name"), (4, "available"), (5, "check"),
+                   (6, "update"), (8, "summary_status")]
+
+        def cell_data_func(_column, cell, model, it, data):
+            # Get the object from the model
+            obj = model.get_value(it, data)
+            # Set the cell value to the name of the object
+            cell.set_property("markup", str(obj))
+
+        for col, name in headers:
+            renderer = self.builder.get_object(name + "_renderer")
+            column = self.builder.get_object(name + "_column")
+            column.set_cell_data_func(renderer, cell_data_func, col)
+            renderer.props.ypad = 10
+            if not name.endswith("name") and name != "summary_status":
+                # center
+                renderer.props.xalign = 0.5
+
+        self.main_window.connect("delete-event", self.window_close)
+        self.main_window.connect("key-press-event", self.check_escape)
+
+        self.load_css()
+
+        self.main_window.show_all()
+
+    @staticmethod
+    def load_css():
+        style_provider = Gtk.CssProvider()
+        css = b'''
```

Comment:
let's put this in a separate file... it's fine now, but I can see a future in which we have 100 lines of CSS embedded in our python code :D